src/ast/passes/macro_expansion.cpp

```diff
@@ -223,29 +236,40 @@ void MacroExpander::visit(Statement &stmt)
     return;
   }
```

Comment:
Aside: why is this duplicated? `ExprStatement` has an `Expression` that will be visited. Do you actually need a visitor for `Statement` at all?

Reply:
I think I can do some consolidation but let me look at it in a follow-up PR. Basically we need to know if the parent of the expression is an expression statement because we're not allowed to replace this
`$x = my_macro();` with this `$x = { $y = 1; }` or `$x = $y = 1;`, which are expression statements.

Reply:
- https://github.com/bpftrace/bpftrace/pull/4426